MeshGeoToolsLib/GeoMapper.cpp

```diff
@@ -115,34 +140,54 @@ void GeoMapper::mapData()
 		min_val = bounding_box.getMinPoint()[2];
 		max_val = bounding_box.getMaxPoint()[2];
 	}
-	std::size_t nPoints (points->size());
 
-	if (!is_borehole)
+	for (auto * pnt : points)
 	{
-		for (unsigned j=0; j<nPoints; ++j)
+		double offset =
+		    (_grid)
+		        ? (getMeshElevation((*pnt)[0], (*pnt)[1], min_val, max_val) -
+		           (*pnt)[2])
+		        : getDemElevation(*pnt);
+
+		if (!GeoLib::isBorehole(pnt))
+			continue;
+		GeoLib::StationBorehole* borehole = static_cast<GeoLib::StationBorehole*>(pnt);
+		const std::vector<GeoLib::Point*> layers = borehole->getProfile();
```

Comment:
Could be `... = pnt->getProfile();` and then the `borehole` pointer is not needed.